server/src/main/java/com/google/musicanalysis/site/AnalysisServlet.java

```diff
@@ -57,11 +58,16 @@ protected void doGet(HttpServletRequest req, HttpServletResponse res)
     NameAndChannel videoInfo = getNameAndChannel(nameJson);
 
     HashMap<String, String> perspectiveMap = analyzeWithPerspective(cumulativeComments);
-    NLPResult commentsSentiment = analyzeWithNLP(cumulativeComments);
+    
+    HashMap<Integer, NLPResult> unweightedNLPMap = new HashMap<>();
+    for (CommentLikes commentAndLikes: commentArray) {
+        unweightedNLPMap.put(commentAndLikes.likes, analyzeWithNLP(commentAndLikes.comment));
```

Comment:
Your total number of likes won't be unique. Instead make your key the NLPResult and your key's value the total number of likes. Otherwise you'll likely get collisions from comments with the same number of likes (which includes many comments with 1 or 0 likes).